Clear up common questions releases (#47394)
rails/rails: guides/source/maintenance_policy.md

In special situations, where someone from the Core Team agrees to support more series,
they are included in the list of supported series.

For unsupported series, bug fixes may coincidentally land in a stable branch,

> but are unreleased in an official version. It is recommended to point your

Reviewer: ```suggestion
but won't be released in an official version. It is recommended to point your
```

> application at the stable branch using git.

Reviewer: ```suggestion
application at the stable branch using Git for unsupported versions.
```

When a release series is no longer supported, it's your own responsibility to
deal with bugs and security issues. We may provide backports of the fixes and

> publish them to git, however there will be no new versions released. We

Reviewer: ```suggestion
merge them, however there will be no new versions released. We
```

> recommend to point your application at the stable branch using git. If you are

Reviewer: ```suggestion
recommend to point your application at the stable branch using Git. If you are
```
